lib/elixir/lib/enum.ex

```diff
@@ -2532,6 +2532,156 @@ defmodule Enum do
     end
   end
 
+  @doc """
+  Pulls out either a single element (denoted by an integer index) or a contiguous range of
+  the enumerable (given by a range) and inserts it in front of the value previously at the
+  insertion index.
+
+  The semantics of the range to be moved match the semantics of Enum.slice/2. Specifically,
+  that means:
+
+  * Indices are normalized, meaning that negative indexes will be counted from the end
+    (for example, -1 means the last element of the enumerable). This will result in *two*
+    traversals of your enumerable on types like lists that don't provide a constant-time count.
+  * If the normalized index range's `last` is out of bounds, the range is truncated to the last element.
+  * If the normalized index range's `first` is out of bounds, the selected range for sliding
+    will be empty, so you'll get back your input list.
+  * Decreasing ranges (e.g., the range 5..0//1) also select an empty range to be moved, so you'll
+    get back your input list.
+  * Ranges with any step but 1 will raise an error.
+
+  ## Examples
+
+      # Slide a single element
+      iex> Enum.slide([0, 1, 2, 3, 4, 5, 6], 5, 1)
+      [0, 5, 1, 2, 3, 4, 6]
+
+      # Slide a range of elements backward
+      iex> Enum.slide([0, 1, 2, 3, 4, 5, 6], 3..5, 1)
+      [0, 3, 4, 5, 1, 2, 6]
+
+      # Slide a range of elements forward
+      iex> Enum.slide([0, 1, 2, 3, 4, 5, 6], 1..3, 5)
+      [0, 4, 5, 1, 2, 3, 6]
+
+      # Slide with negative indices (counting from the end)
+      iex> Enum.slide([0, 1, 2, 3, 4, 5, 6], 3..-1//1, 2)
+      [0, 1, 3, 4, 5, 6, 2]
+      iex> Enum.slide([0, 1, 2, 3, 4, 5, 6], -4..-2, 1)
+      [0, 3, 4, 5, 1, 2, 6]
```

Comment:
I just realized something, by having the input list be numbers, someone can think that 1..3 is referring to the actual values and not the position. Shall we use `[:a, :b, :c, ...]` as input?

Reply:
Makes sense!